### PR TITLE
Multi-GPU training for our autoencoder model

### PR DESCRIPTION
**Pull request by jsjason**
This PR extends the original training script to support multi-gpu training.
We simply define the same model on each gpu (while sharing variables, of course), extract losses from each gpu, and calculate gradients on the average loss.
```python
avg_loss = tf.reduce_mean(losses, 0)
...
train_op = slim.learning.create_train_op(avg_loss, optimizer, ...)
```
One should make separate training dataset file for each gpu. More specifically, filename `0` ~ `n-1` should exist under `train_path` directory, where `n` is number of gpus to use.

### CODE REVIEW

**Comment by gyeongin** [comment]
LGTM.
